app/src/main/java/ch/sdp/vibester/auth/FireBaseAuthenticator.kt

```diff
@@ -25,17 +29,58 @@ class FireBaseAuthenticator @Inject constructor() {
          * API: return the mail of the user if logged in otherwise empty string
          */
         fun getCurrentUserMail(): String {
-            if (isLoggedIn()) {
+            if (isLoggedIn() || TestMode.isTest()) {
                 return FirebaseAuth.getInstance().currentUser!!.email.toString()
             } else {
                 return ""
             }
         }
 
+        fun getCurrentUID(): String {
+            if ( (isLoggedIn() || TestMode.isTest()) && !TestMode.isTest2()) {
+                return FirebaseAuth.getInstance().currentUser!!.uid
+            } else {
+                return ""
+            }
+        }
+
+
+        /**
+         * A function to return the result of google sign in
+         * @param requestCode a request code
+         * @param resultCode a result code
+         * @param data intent returned from google sign in
+         */
+        fun googleActivityResult(requestCode: Int, resultCode: Int, data: Intent?, ctx: Context): String? {
+            if (requestCode == 1000) {
+                try {
+                    val account = GoogleSignIn.getSignedInAccountFromIntent(data).result
+                    val client = Identity.getSignInClient(ctx)
+                    val idToken = client.getSignInCredentialFromIntent(data).googleIdToken
+                    val ficred = GoogleAuthProvider.getCredential(idToken, null)
+                    FirebaseAuth.getInstance().signInWithCredential(ficred)
+                    return account.email
+                } catch (e: Exception) {
+                    return "Authentication error"
+                }
+            } else {
+                return "Authentication error"
```

Comment:
The one line 67 is a real error that should never happens ! It is if we use that result listener with another requestCode (means for another request type).
Then at line 64, it can be a problem of internet connection or an already existing user. Would need much more time to split up differents error.